Build failure: pwvucontrol

### Steps To Reproduce

Steps to reproduce the behavior:
1. build with nixpkgs unstable 

### Build log

```
error: builder for '/nix/store/klip33d1771r9j155438gjkryi8j51ms-pwvucontrol-0.2.drv' failed with exit code 1;
       last 10 log lines:
       >   Package wireplumber-0.4 was not found in the pkg-config search path.
       >   Perhaps you should add the directory containing `wireplumber-0.4.pc'
       >   to the PKG_CONFIG_PATH environment variable
       >   No package 'wireplumber-0.4' found
       >
       > warning: build failed, waiting for other jobs to finish...
       > [6/7] Merging translations for data/com.saivert.pwvucontrol.metainfo.xml
       > FAILED: src/pwvucontrol
       > /nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/env CARGO_HOME=/build/source/build/cargo-home CODEGEN_BUILD_DIR=/build/source/build/src /nix/store/nhcidacc2bad8jaq965i3bwwnf777b6r-cargo-1.76.0/bin/cargo build --manifest-path /build/source/Cargo.toml --target-dir /build/source/build/src --release && cp src/release/pwvucontrol src/pwvucontrol
       > ninja: build stopped: subcommand failed.
```

### Additional context

seems to be a similar issue that was just fixed with waybar [here](https://github.com/NixOS/nixpkgs/pull/297158)

### Notify maintainers

<!--
@figsoda
-->

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.9-xanmod1, NixOS, 24.05 (Uakari), 24.05.20240319.b06025f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/zaza7mgggz4m5h6z18kajabf4wciaj47-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/audio/pwvucontrol/default.nix
+++ b/pkgs/applications/audio/pwvucontrol/default.nix
@@ -1,6 +1,7 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, fetchFromGitLab
 , cargo
 , desktop-file-utils
 , meson
@@ -19,22 +20,33 @@
 , wireplumber
 }:
 
+let
+  wireplumber_0_4 = wireplumber.overrideAttrs (attrs: rec {
+    version = "0.4.17";
+    src = fetchFromGitLab {
+      domain = "gitlab.freedesktop.org";
+      owner = "pipewire";
+      repo = "wireplumber";
+      rev = version;
+      hash = "sha256-vhpQT67+849WV1SFthQdUeFnYe/okudTQJoL3y+wXwI=";
+    };
+  });
+in
 stdenv.mkDerivation rec {
   pname = "pwvucontrol";
-  version = "0.2";
+  version = "0.3.1";
 
   src = fetchFromGitHub {
     owner = "saivert";
     repo = "pwvucontrol";
     rev = version;
-    hash = "sha256-jBvMLewBZi4LyX//YUyJQjqPvxnKqlpuLZAm9zpDMrA=";
+    hash = "sha256-t4JUpKtuO1JgrdsNbl0M/SwPmoqu7WxsaD74vlJeVhY=";
   };
 
   cargoDeps = rustPlatform.importCargoLock {
     lockFile = ./Cargo.lock;
     outputHashes = {
-      "libspa-0.6.0" = "sha256-CVLQ9JXRMo78/kay1TpRgRuk5v/Z5puPVMzLA30JRk8=";
-      "wireplumber-0.1.0" = "sha256-wkku9vqIMdV+HTkWCPXKH2KM1Xzf0xApC5zrVmgxhsA=";
+      "wireplumber-0.1.0" = "sha256-+LZ8xKok2AOegW8WvfrfZGXuQB4xHrLNshcTOHab+xQ=";
     };
   };
 
@@ -58,7 +70,7 @@ stdenv.mkDerivation rec {
     libadwaita
     pango
     pipewire
-    wireplumber
+    wireplumber_0_4
   ];
 
   meta = with lib; {

